backend/routes/goals.py

```
@@ -278,12 +271,50 @@ def should_show_personal_goal_button():
+        # TODO: this should also never occur
+        if today_activity.gems_earned is None:
+            today_activity.gems_earned = 0
+            logger.debug(f"today activity gems was None, set to 0")
+
+        today_activity.gems_earned += GEMS_FOR_WEEKLY_COMPLETION_QUOTA
+        user.gems += GEMS_FOR_WEEKLY_COMPLETION_QUOTA
+        db.session.commit()
+
+        logger.info("Gems for weekly quota goal added successfully")
+        return jsonify({"message": "Gems added successfully"}), 200
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)